### ISSUE TITLE
(v0.12.16) Validation failing with operation-2xx-response incorrectly

### ISSUE THREAD

**Issue by kf6kjg** [Redocly/openapi-cli#145]
Given the below YAML the command `npx openapi validate src/test/openapi.yaml` throws this warning:

```
[1] src/test/openapi.yaml:8:3 at #/components/responses
Operation must have at least one 2xx response.
```

```yaml
---
openapi: 3.0.3
info:
  title: Testing API
  version: 1.0.0

components:
  responses:
    RPCSuccessMessage:
      description: Halo whirl'd.
      content:
        application/json:
          schema:
            type: object

paths:
  /:
    post:
      operationId: successfulJsonResponse
      responses:
        200:
          $ref: "#/components/responses/RPCSuccessMessage"
      tags:
        - Testing

tags:
  - name: Testing
```

From my reading of the OpenAPI spec,
* `#/components` is a `Components Object`
* `#/components/responses` is a `Map[string, Response Object | Reference Object]`

The warning makes it seem like `#/components/responses` is being treated as a `Responses Object` instead of as the correct `Map[string, Response Object | Reference Object]`.

**Comment by rohit-gohri** [comment]
I'm having the same problem

**Comment by RomanHotsiy** [comment]
Thanks for reporting! We are awara. This will be fixed as part of https://github.com/Redocly/openapi-cli/pull/146

**Comment by adamaltman** [comment]
This is confirmed fixed by #146 in `1.0.0-beta.1`.

You will notice errors, but they aren't the `2xx` issue, and it's related to falling back to our `recommended` configuration by default.

```
➜  bugs NO_COLOR=1 openapi lint issue-145.yaml
No configurations were defined in extends -- using built in recommended configuration by default.

validating issue-145.yaml...
[1] issue-145.yaml:1:1 at #/

Servers must be present.

 1 | openapi: 3.0.3
   | ^^^^^^^^^^^^^^
 2 | info:
   | ^^^^^
 … | < 23 more lines >
26 |   - name: Testing
   |   ^^^^^^^^^^^^^^^
27 |

Error was generated by the no-empty-servers rule.


[2] issue-145.yaml:17:5 at #/paths/~1

`/` should not have a trailing slash.

15 | paths:
16 |   /:
17 |     post:
   |     ^^^^^
18 |       operationId: successfulJsonResponse
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 … |       < 5 more lines >
24 |
   | ^
25 | tags:
26 |   - name: Testing

Error was generated by the no-path-trailing-slash rule.


[3] issue-145.yaml:18:7 at #/paths/~1/post

Operation object should contain `summary` field.

16 |   /:
17 |     post:
18 |       operationId: successfulJsonResponse
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
19 |       responses:
   |       ^^^^^^^^^^
 … |       < 4 more lines >
24 |
   | ^
25 | tags:
26 |   - name: Testing

Error was generated by the operation-summary rule.


[4] issue-145.yaml:3:3 at #/info

Info object should contain `description` field.

1 | openapi: 3.0.3
2 | info:
3 |   title: Testing API
  |   ^^^^^^^^^^^^^^^^^^
4 |   version: 1.0.0
  |   ^^^^^^^^^^^^^^
5 |
6 | components:

Warning was generated by the info-description rule.


[5] issue-145.yaml:3:3 at #/info

Info object should contain `license` field.

1 | openapi: 3.0.3
2 | info:
3 |   title: Testing API
  |   ^^^^^^^^^^^^^^^^^^
4 |   version: 1.0.0
  |   ^^^^^^^^^^^^^^
5 |
6 | components:

Warning was generated by the info-license rule.


[6] issue-145.yaml:26:5 at #/tags/0

Tag object should contain `description` field.

24 |
25 | tags:
26 |   - name: Testing
   |     ^^^^^^^^^^^^^
27 |

Warning was generated by the tag-description rule.


issue-145.yaml: validated in in 46ms

❌ Validation failed with 3 errors and 3 warnings.

run with `--generate-ignore-file` to add all problems to ignore file.
```

**Comment by RomanHotsiy** [comment]
```
[2] issue-145.yaml:17:5 at #/paths/~1

`/` should not have a trailing slash.

15 | paths:
16 |   /:
17 |     post:
   |     ^^^^^
18 |       operationId: successfulJsonResponse
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 … |       < 5 more lines >
24 |
   | ^
25 | tags:
26 |   - name: Testing

Error was generated by the no-path-trailing-slash rule.
```

I think this is another issue. I've opened a separate issue: https://github.com/Redocly/openapi-cli/issues/161
